features/mbedtls/platform/COMPONENT_PSA_SRV_IMPL/src/default_random_seed.cpp

```diff
@@ -0,0 +1,19 @@
+
+#include "mbed.h"
+#include "default_random_seed.h"
+#include "psa_prot_internal_storage.h"
+
+int mbed_default_seed_read( unsigned char *buf, size_t buf_len )
+{
+	psa_its_status_t rc = psa_its_get(MBED_RANDOM_SEED_ITS_UID, 0, buf_len, buf);
+	return rc;
```

Comment:
Simply returning `rc` isn't going to work when there is an error. PSA ITS errors are positive integers. The code checking for errors from `mbedtls_nv_seed_read()` interprets values less than zero as errors. This will mean that no entropy was provided (due to the error), but the system will think entropy was provided (because a non-negative value was returned): catastrophic for security!